test/browser-tests/rebind.js

```diff
@@ -350,3 +350,54 @@ test( 'Computations are not triggered prematurely on rebind (#2259)', t => {
 
 	ractive.splice( 'items', 0, 1 );
 });
+
+test( 'rebinds of IF sections do not add new context (#2454)', function ( t ) {
+	const ractive = new Ractive({
+		el: fixture,
+		template: `
+{{#items}}
+	{{#disabled}} 
+		{{this.name}} 
+	{{/}}
+{{/}}`,
+		data: {
+			items: [
+				{ name: "foo", disabled: true },
+				{ name: "bar", disabled: true }
+			]
+		}
+	});
+
+	t.htmlEqual( fixture.innerHTML, 'foobar' );
+
+	// Remove first item and test that the second item is still present, and not a new context.
+	ractive.splice( 'items', 0, 1 );
+
+	t.htmlEqual( fixture.innerHTML, 'bar' );
+});
+
+test( 'rebinds of UNLESS sections do not add new context (#2454)', function ( t ) {
+	const ractive = new Ractive({
+		el: fixture,
+		template: `
+{{#items}}
+	{{#disabled}} 
```

Comment:
Thanks, I've renamed the unless test to if/else, since that is what it tests, and added a proper unless test -> {{^disabled}}